codeSnippets/snippets/auth-oauth-google/src/main/kotlin/com/example/Application.kt

```diff
@@ -0,0 +1,63 @@
+package com.example
+
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.http.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+
+fun Application.main() {
+    val httpClient = HttpClient(CIO) {
+        install(JsonFeature)
+    }
+    install(Authentication) {
+        oauth("auth-oauth-google") {
+            urlProvider = { "http://localhost:8080/callback" }
+            providerLookup = {
+                OAuthServerSettings.OAuth2ServerSettings(
+                    name = "google",
+                    authorizeUrl = "https://accounts.google.com/o/oauth2/auth",
+                    accessTokenUrl = "https://accounts.google.com/o/oauth2/token",
+                    requestMethod = HttpMethod.Post,
+                    clientId = System.getenv("GOOGLE_CLIENT_ID"),
+                    clientSecret = System.getenv("GOOGLE_CLIENT_SECRET"),
+                    defaultScopes = listOf("https://www.googleapis.com/auth/userinfo.profile")
+                )
+            }
+            client = httpClient
+        }
+    }
+    routing {
+        var principal: OAuthAccessTokenResponse.OAuth2? = null
```

Comment:
That's not a right way to do it. You should save an access token in a user's session.

Reply:
ok, saved a token to cookies